crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -870,6 +871,30 @@ impl Timeline {
             Ok(false)
         }
     }
+
+    pub async fn pin_event(&self, event_id: &EventId) -> Result<bool> {
+        let mut pinned_events = self.room().pinned_events().get();
+        let event_id = event_id.to_owned();
```

Comment:
I think you can do the `to_owned()` inside the `else` here, saving a copy.

Reply:
I tried to do that but since `pinned_events.contains` requires a `OwnedEventId`, I would need to do `to_owned()` also in the `contains` which would make for two copies

Reply:
Cool, ignore me then!